### PR TITLE
Verify torrent category on upload

### LINKED ISSUE #682: Torrents with invalid categories.
Example: https://nyaa.pantsu.cat/view/924512

There's no checking if the torrent category on the form actually exists, it checks only if it is in the format `x_y`, where `x` and `y` are both numbers. You can reproduce this locally by editing the `value` field of the selected `option` on the upload page, and trying to upload.

### PR DESCRIPTION
**Pull request by ElegantMonkey**
Should fix #682

### CODE REVIEW

File: router/template_functions.go [hunk 1]
```diff
@@ -161,51 +162,14 @@ var FuncMap = template.FuncMap{
...
-            case "5_11":    e = "live_action_raw"
-            case "6_":      e = "pictures"
-            case "6_15":    e = "pictures_graphics"
-            case "6_16":    e = "pictures_photos"
-            case "1_":      e = "software"
-            case "1_1":     e = "software_applications"
-            case "1_2":     e = "software_games"
-        }
-        return e
+    "CategoryID": func(category string, sub_category string) string {
```
Comment (sfan5) [inline router/template_functions.go]: should probably be `CategoryName`

**Comment by sfan5** [comment]
Also you're missing the case where people could try to post in a parent category (without a subcat selected)

**Comment by akuma06** [comment]
@sfan5 Shouldn't it work like that? If someone post in Anime Category, it should check "3_". It exists so it uploads

File: router/template_functions.go (hunk 1 above)
Comment (akuma06) [inline router/template_functions.go]: I agree, it would be more fitted to the function

**Comment by ElegantMonkey** [comment]
@sfan5 @akuma06 The upload handler splits the category on the _ sign, and try to convert each string part to an int. The current behavior fails with, say, `3_`, because splitting would return `["3", ""]` and Atoi would fail to convert an empty string.

Unless you mean people should be able to post on a parent category, in which this should be fixed.

**Comment by sfan5** [comment]
Oh, then that's okay already.
@akuma06 No, torrents should always have a sub-category attached.

**Pull request merged**